### PR TITLE
Added warning about keeping empty tables on first filter call

### PR DESCRIPTION
**Pull request by sanderson**
- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md [hunk 1]
```diff
@@ -49,13 +49,19 @@ Defines the behavior for empty tables.
 Potential values are `keep` and `drop`.
 Defaults to `drop`.
 
+_**Data type:** String_
+
 ##### drop
-Empty tables are dropped.
+Tables left without rows are dropped.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md]: Tables w/o rows?

File: content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md [hunk 2]
```diff
@@ -49,13 +49,19 @@ Defines the behavior for empty tables.
...
 
 ##### keep
-Empty tables are output to the next transformation.
+Tables left without rows are output to the next transformation.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md]: same

File: content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md [hunk 3]
```diff
@@ -49,13 +49,19 @@ Defines the behavior for empty tables.
...
 
-_**Data type:** String_
+{{% warn %}}
+Keeping empty tables with your first `filter()` function can have severe performance costs.
+For higher performance, use your first `filter()` function to do basic filtering,
+then keep empty tables on subsequent `filter()` calls.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md]: curious, why keep empty tables on subsequent filter()s..maybe add example?

File: content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md [hunk 4]
```diff
@@ -52,15 +52,17 @@ Defaults to `drop`.
...
-Tables left without rows are output to the next transformation.
+Tables without rows are output to the next transformation.
 
 {{% warn %}}
-Keeping empty tables with your first `filter()` function can have severe performance costs.
+Keeping empty tables with your first `filter()` function can have severe performance
+costs since it retains empty tables from your entire data set.
 For higher performance, use your first `filter()` function to do basic filtering,
-then keep empty tables on subsequent `filter()` calls.
+then keep empty tables on subsequent `filter()` calls with smaller data sets.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md]: would folks do this, for example, to accurately aggregate data? If empty data points aren't included in the output, could you count the total? other use cases?

Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/built-in/transformations/filter.md]: That's one of the use cases, yes. This is covered in the `count()` doc.

**Pull request merged**